### DISCUSSION TITLE
TypeError: Failed to set the 'files' property on 'HTMLInputElement': Failed to convert value to 'FileList'.

### DISCUSSION THREAD

**Discussion by jb-thery** [react-hook-form/react-hook-form#5164] [Q&A]
Hello

I create a little cms with react hook form, but I have an issue with images.
If I refresh the browser I get this error :

```
TypeError: Failed to set the 'files' property on 'HTMLInputElement': Failed to convert value to 'FileList'.
```

I have no Idea, where it's come

Codesandbox :
https://codesandbox.io/s/pensive-visvesvaraya-m3ifv

**Comment by bluebill1049** [comment]
how can i reproduce this? can you provide an isolated codesandbox as well?

**Reply by jb-thery** [reply]
Hi, 

Codesandbox:

https://codesandbox.io/s/pensive-visvesvaraya-m3ifv

To see the error, Go to the codesandbox and refresh the preview browser

**Comment by thanh-nguyen-95** [comment]
> Hello
> 
> I create a little cms with react hook form, but I have an issue with images.
> If I refresh the browser I get this error :
> 
> ```
> TypeError: Failed to set the 'files' property on 'HTMLInputElement': Failed to convert value to 'FileList'.
> ```
> 
> I have no Idea, where it's come
> 
> Codesandbox :
> https://codesandbox.io/s/pensive-visvesvaraya-m3ifv

I don't see the error you mentioned in the CodeSandbox. All I see is an empty error.

Anyway, I had a quick look at the code. The error might come from using `<FaFileImageInput>`. I think the initial value is `undefined` or a `string`, which causes the error when converting to `FileList`.

**Comment by jb-thery** [comment]
@thanh-nguyen-95 @bluebill1049 

Thanks, what it's the good initial value for image input ?

![https://i.ibb.co/xgPzQd1/Capture-d-e-cran-2021-05-12-a-10-26-16.png](https://i.ibb.co/xgPzQd1/Capture-d-e-cran-2021-05-12-a-10-26-16.png)

**Reply by thanh-nguyen-95** [reply]
I wouldn't put the `value` to an `<input type="file" />`.

I think you should separate the `value` and `onChange` into 2 seperate part:

- `value`: displayed as an image (a preview) when it's defined
- `onChange`: act as a setter when a file is upload

When use with `Controller`, it should look like this:

```jsx
<Controller
  name="yourimage"
  control={control}
  defaultValue=""
  render={({ field }) => (
    <>
      {field.value && <img src={field.value} />}
      <input
        type="file"
        accept="image/*"
        onChange={(event) => {
          // read the file as base64 using FileReader
          // when you get the result, simply use `field.onChange(imageAsBase64)` to update the value         
        }}
      />
    </>
  )}
/>
``` 

You can use the `<label>` with hidden `<input type="file" />` trick if you want something more fancy.

This is just an suggestion, you can implement it however you want.

**Reply by bluebill1049** [reply]
thank you @thanh-nguyen-95 ❤️

**Reply by jb-thery** [reply]
Yes thank you very much I will try it asap, have a nice day

**Reply by jb-thery** [reply]
@bluebill1049 thanks you for this amazing react hook from !
